src/renderer/Window.cpp

```diff
@@ -178,55 +180,67 @@ Window::Window(const string &title, const Settings::Section section, const strin
+      SDL_SetStringProperty(props, SDL_PROP_WINDOW_CREATE_TITLE_STRING, title.c_str());
+      SDL_SetNumberProperty(props, SDL_PROP_WINDOW_CREATE_X_NUMBER, wnd_x);
+      SDL_SetNumberProperty(props, SDL_PROP_WINDOW_CREATE_Y_NUMBER, wnd_y);
+      SDL_SetNumberProperty(props, SDL_PROP_WINDOW_CREATE_WIDTH_NUMBER, m_width);
+      SDL_SetNumberProperty(props, SDL_PROP_WINDOW_CREATE_HEIGHT_NUMBER, m_height);
+      SDL_SetNumberProperty(props, SDL_PROP_WINDOW_CREATE_FLAGS_NUMBER, wnd_flags);
+#if defined(ENABLE_BGFX) && defined(__ANDROID__)
+      SDL_SetBooleanProperty(props, SDL_PROP_WINDOW_CREATE_EXTERNAL_GRAPHICS_CONTEXT_BOOLEAN, true);
+#endif
+      m_nwnd = SDL_CreateWindowWithProperties(props);
+      SDL_DestroyProperties(props);
 
 #ifdef __LIBVPINBALL__
-      SDL_SysWMinfo wmInfo;
-      SDL_VERSION(&wmInfo.version);
-      SDL_GetWindowWMInfo(m_nwnd, &wmInfo);
-      VPinballLib::WindowCreatedStruct windowCreatedStruct = { (void*)wmInfo.info.uikit.window, title.c_str() };
+      props = SDL_GetWindowProperties(m_nwnd);
+      VPinballLib::WindowCreatedStruct windowCreatedStruct = { (void*)SDL_GetPointerProperty(props, SDL_PROP_WINDOW_UIKIT_WINDOW_POINTER, NULL), title.c_str() };
       VPinballLib::VPinball::SendEvent(VPinballLib::Event::WindowCreated, &windowCreatedStruct);
 #endif
 
-      SDL_DisplayMode mode;
+      const SDL_DisplayMode* mode;
+
       if (m_fullscreen)
       {
-         SDL_GetWindowDisplayMode(m_nwnd, &mode);
-         m_refreshrate = mode.refresh_rate;
+         mode = SDL_GetWindowFullscreenMode(m_nwnd);
+         m_refreshrate = mode ? mode->refresh_rate : 0;
          m_screenwidth = m_width;
          m_screenheight = m_height;
-         if (fsRefreshRate > 0 && validatedFSRefreshRate != m_refreshrate) // Adjust refresh rate if needed
+         // JSM174 if (fsRefreshRate > 0 && validatedFSRefreshRate != m_refreshrate) // Adjust refresh rate if needed
          {
-            Uint32 format = mode.format;
+            Uint32 format = mode ? mode->format : 0;
             bool found = false;
-            for (int index = 0; index < SDL_GetNumDisplayModes(m_adapter); index++)
-            {
-               SDL_GetDisplayMode(m_adapter, index, &mode);
-               if ((mode.w == m_width) && (mode.h == m_height) && (mode.refresh_rate == validatedFSRefreshRate) && (mode.format == format))
-               {
-                  SDL_SetWindowDisplayMode(m_nwnd, &mode);
-                  m_refreshrate = validatedFSRefreshRate;
-                  found = true;
-                  break;
+
+            int num_modes = 0;
+            SDL_DisplayMode **modes = SDL_GetFullscreenDisplayModes(m_adapter, &num_modes);
+            if (modes) {
+               for (int index = 0; index < num_modes; ++index) {
+                 mode = modes[index];
+                 // JSM174 if ((mode->w == m_width) && (mode->h == m_height) && (mode->refresh_rate == validatedFSRefreshRate) && (mode->format == format))
```

Comment:
What is unclear here? Cause this is definetly necessary to get the right resolution, etc for fullscreen, right?

Reply:
for some reason, `mode = SDL_GetWindowFullscreenMode(m_nwnd);` and `mode = SDL_GetDesktopDisplayMode(m_display);` always come back as `NULL`, so I was getting into asserts.

I figure maybe if you or @vbousquet attempt to try running on windows, whatever fixes are needed, will ultimately fix this.